File: app/src/main/java/com/hundaol/ethiopic/cal/ICal.java
package com.hundaol.ethiopic.cal;

/**
 * Created by jmpirie on 2017-04-14.
 */
public interface ICal {

    // initialization

    int today();

    int fromDate(int year, int month, int day);

    // date

    int getYear(int jdn);

    int getMonth(int jdn);

    int getDay(int jdn);
    
    // day

    int nextDay(int jdn);

    int nextDay(int jdn, int days);

    int prevDay(int jdn);

    int prevDay(int jdn, int days);

    // week

    int getDayOfWeek(int jdn);

    int getFirstDayOfWeek(int jdn);

    int getLastDayOfWeek(int jdn);

    boolean isWeekend(int jdn);

    boolean isWeekday(int jdn);

    // month

    int firstOfMonth(int jdn);

    int lastOfMonth(int jdn);

    int nextMonth(int jdn);

    int prevMonth(int jdn);

    char[] getMonthName(int jdn);

    int getDaysInMonth(int jdn);

    int getWeeksInMonth(int jdn);

    int getWeekOfMonth(int jdn);

    int getDayOfWeekOfFirstDayInMonth(int jdn);

    int getDayOfWeekOfLastDayInMonth(int jdn);

    int getLastFullWeek(int jdn);

    // year

    int getMonthsInYear();

    // week number

    int getWeekNumber(int jdn);
}

File: app/src/main/java/com/hundaol/ethiopic/cal/ICal.kt
package com.hundaol.ethiopic.cal

/**
 * Created by jmpirie on 2017-04-14.
 */
interface ICal {

    // initialization

    fun today(): Int

    fun fromDate(year: Int, month: Int, day: Int): Int

    // date

    fun getYear(jdn: Int): Int

    fun getMonth(jdn: Int): Int

    fun getDay(jdn: Int): Int

    // day

    fun nextDay(jdn: Int): Int

    fun nextDay(jdn: Int, days: Int): Int

    fun prevDay(jdn: Int): Int

    fun prevDay(jdn: Int, days: Int): Int

    // week

    fun getDayOfWeek(jdn: Int): Int

    fun getFirstDayOfWeek(jdn: Int): Int

    fun getLastDayOfWeek(jdn: Int): Int

    fun isWeekend(jdn: Int): Boolean

    fun isWeekday(jdn: Int): Boolean

    // month

    fun firstOfMonth(jdn: Int): Int

    fun lastOfMonth(jdn: Int): Int

    fun nextMonth(jdn: Int): Int

    fun prevMonth(jdn: Int): Int

    fun getMonthName(jdn: Int): CharArray

    fun getDaysInMonth(jdn: Int): Int

    fun getWeeksInMonth(jdn: Int): Int

    fun getWeekOfMonth(jdn: Int): Int

    fun getDayOfWeekOfFirstDayInMonth(jdn: Int): Int

    fun getDayOfWeekOfLastDayInMonth(jdn: Int): Int

    fun getLastFullWeek(jdn: Int): Int

    // year

    val monthsInYear: Int

    // week number

    fun getWeekNumber(jdn: Int): Int
}